src/Node/ForNode.php

```diff
@@ -25,8 +25,12 @@ class ForNode extends Node
 {
     protected $loop;
 
-    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, AbstractExpression $ifexpr = null, \Twig_NodeInterface $body, \Twig_NodeInterface $else = null, $lineno, $tag = null)
+    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, AbstractExpression $ifexpr = null, \Twig_NodeInterface $body = null, \Twig_NodeInterface $else = null, $lineno = 0, $tag = null)
     {
+        if (null === $body) {
```

Comment:
This would be unnecessary if you bump the min version to 7.1 instead (7.0 reached end of support at the same time than 5.6, so the argument allowing to drop support for PHP 5.x also applies to 7.0), as you will be able to change `$ifExpr` to use `?AbstractExpression $ifExpr` instead.

Reply:
btw, if we go this way, I suggest keeping the lineno mandatory in all nodes by using the same way. We don't want nodes to be created without specifying their line, as that would cause issues for error reporting.

Reply:
Symfony 3.4 is still compatible with PHP 7.0.

Reply:
PHP 7.0 dropped